tests/tests/test_train_schedule.py

```diff
@@ -301,6 +304,213 @@ def test_etcs_schedule_result_stop_brake_from_mrsp(etcs_scenario: Scenario, etcs
+                    {"id": "last", "track": "TH1", "offset": 3_922_000},
+                ],
+                "schedule": [
+                    {"at": "zero", "stop_for": "P0D"},
+                    {"at": "first", "stop_for": "PT10S"},
+                    {"at": "last", "stop_for": "P0D"},
+                ],
+                "margins": {"boundaries": [], "values": ["0%"]},
+                "initial_speed": 0,
+                "comfort": "STANDARD",
+                "constraint_distribution": "STANDARD",
+                "speed_limit_tag": "foo",
+                "power_restrictions": [],
+            }
+        ],
+    )
+
+    schedule = ts_response.json()[0]
+    schedule_id = schedule["id"]
+    ts_id_response = requests.get(f"{EDITOAST_URL}train_schedule/{schedule_id}/")
+    ts_id_response.raise_for_status()
+    simu_response = requests.get(
+        f"{EDITOAST_URL}train_schedule/{schedule_id}/simulation?infra_id={etcs_scenario.infra}"
+    )
+    simulation_final_output = simu_response.json()["final_output"]
+
+    assert len(simulation_final_output["positions"]) == len(simulation_final_output["speeds"])
+
+    # To debug this test: please add a breakpoint then use front to display speed-space chart
+    # (activate Context for Slopes and Speed limits).
+
+    # This case hits an LoA curve (slowdown of the MRSP), but it's not the point to test it here.
+
+    # Check that the curves does respect the EoA + SvL (EoA = stops), and that there is an
+    # acceleration then deceleration in between (maintain speed when reach the MRSP).
+    # Here, the EoA is at 41038m, and the SvL is at the switch at 41138m.
+    distance_to_svl = 100_000
+    first_stop_offset = SVL_STOP_OFFSET - distance_to_svl
+    final_stop_offset = 45_060_000
+    stop_offsets = [
+        0,
+        first_stop_offset,
+        final_stop_offset,
+    ]
+
+    # Check null speed at stops
+    for stop_offset in stop_offsets:
+        assert _get_current_or_next_speed_at(simulation_final_output, stop_offset) == 0
+
+    # Check only one acceleration then only one deceleration between stops
+    for offset_index in range(1, len(stop_offsets) - 1):
+        accelerating = True
+        prev_speed = 0
+        start_pos_index = bisect.bisect_left(simulation_final_output["positions"], stop_offsets[offset_index - 1])
+        end_pos_index = bisect.bisect_left(simulation_final_output["positions"], stop_offsets[offset_index])
+        for pos_index in range(start_pos_index, end_pos_index):
+            current_speed = simulation_final_output["speeds"][pos_index]
+            if accelerating:
+                if prev_speed > current_speed:
+                    accelerating = False
+            else:
+                assert prev_speed >= current_speed
+            prev_speed = current_speed
+
+    # Check that the braking curve starts and ends at the expected offsets.
+    offset_start_first_brake = START_SVL_BRAKE_ON_PH1_SWITCH - distance_to_svl
```

Comment:
As mentioned, it looks to me like we shouldn't mutualize it, as it's only "lucky" that both EoA and SvL upper curve parts are under GUI influence and therefore separated by the same value as the EoA-SvL distance.
"Hard-documentation" though variable names seems too strong (and hides a lot of complexity about how the curves are computed) IMO.

Also, the values are not enough "test-generic" IMO (same as the `SVL_STOP_OFFSET`).

I wouldn't detail what's happening here at all (or just a comment, if you will).
```suggestion
    offset_start_first_brake = 33_361_530
```

🦜 Ditto for the "same location" test above.